Readme.md

```diff
@@ -13,26 +13,31 @@ protocols, but it is not necessarily limited to this.
 > The `evm_simulation` module allows for simulating _any_ transaction; it is agnostic to protocol. See module's
 > documentation.
 
-To further help solve hard problems in the context of exchanging tokens, the crate provides the ProtoGraph structure,
-which can be queried for chained token exchanges and their parameters. This graph structure evolves over time as
-protocol states are changed by user actions. These changes are captured using events. The implemented protocols are
-aware of the state-mutating events and can transition their state correctly given such events.
-
 ## Currently supported protocols:
 
 - Uniswap V2 and Forks
 - Uniswap V3
+- VM enabled protocols
 
 See also `evm` module which can simulate any transaction.
 
 ## Adding a new Protocol
 
 To add a new protocol, you will need to complete the following high-level steps:
 
+#### Native protocol:
+
 1. Create a protocol state struct that implements the `ProtocolSim` struct.
-2. _(legacy information)_ Add associated events and implement the transition method on the protocol state struct.
+2. Create a tycho decoder for the protocol state: i.e. implement `TryFrom` for `ComponentWithState` to your new 
```

Comment:
Is it `TryFrom` or `TryFromWithBlock`, also we should specify the error type of the implementation here I think.

Reply:
It's `TryFrom`. `TryFromWithBlock is only used by VmPoolState which already has a decoder.

Reply:
Would prefer if we were consistent

Reply:
Okay, I'll do this in a separate PR... Diana is waiting on this one.